subsys/bluetooth/host/audio/bass.c

```diff
@@ -929,17 +948,26 @@ static int bass_mod_src(struct bt_conn *conn, struct net_buf_simple *buf)
 		}
 	}
 
-	internal_state->pa_interval = pa_interval;
+	if (state->req_pa_sync_value != pa_sync) {
+		state->req_pa_sync_value = pa_sync;
+		need_synced = true;
+		state_changed = true;
+	}
 
-	/* If no change to the state don't sync PA */
-	if (!state_changed) {
-		bass_pa_sync(conn, internal_state, pa_sync);
+	internal_state->pa_interval = pa_interval;
 
+	if (need_synced) {
+		/* Terminated BIG first if existed */
 		err = bis_sync_cancel(internal_state);
 		if (err) {
 			BT_WARN("Could not terminate existing BIG %d", err);
 			return BT_GATT_ERR(BT_ATT_ERR_UNLIKELY);
 		}
+
+		/* Terminated PA let's re-sync later */
+		bass_pa_sync_cancel(internal_state);
```

Comment:
> Not sure I agree with this change: If the client writes 0 to the `pa_sync` state, aren't we supposed to stop the `pa_sync`?

The main reason for the modification is that it has been cancelled here, so there is no need to call it repeatedly.

In addition, according to bass_pa_sync literally means synchronization and should not include cancel.